### ISSUE TITLE
Override lib name to link for a *-sys package (Windows, msvc)

### ISSUE THREAD

**Issue by fluxxu** [rust-lang/cargo#3318]
TLDR:
**How to make a crate named foo-sys to link with libfoo.lib instead of foo.lib?**

So I was trying to install diesel_cli(@sgrif) on Windows.
I downloaded Postgres libs from 
https://www.postgresql.org/download/windows/
using the **zip archive** link
I set PQ_LIB_DIR to the path of extracted files. Then run cargo build:
```
note: LINK : fatal error LNK1181: cannot open input file 'pq.lib'
```
Then I discovered there is no pq.lib but a libpq.lib in lib folder.
This is a known issue:
https://github.com/diesel-rs/diesel/issues/487 (@Shtong)
https://github.com/sgrif/pq-sys/issues/3

I want to solve this so I forked pq-sys and modified the build script:
I changed

https://github.com/sgrif/pq-sys/blob/master/build.rs#L19
```
   let lib = "pq";
```

into
```
    let lib = if cfg!(target_os = "windows") {
        "libpq"
    } else {
        "pq"
    };
```
and removed 
```
links = "pq"
```
from
https://github.com/sgrif/pq-sys/blob/master/Cargo.toml#L8

Still, no luck because cargo still added pq.lib into build command:
```
link.exe ...
"pq.lib" "libpq.lib" "msvcrt.lib" "ws2_32.lib" "userenv.lib" "shell32.lib" "advapi32.lib" "msvcrt.lib"
  = note: LINK : fatal error LNK1181: cannot open input file 'pq.lib'
```

**Now it links with both "pq.lib" and "libpq.lib".** 

Then I discovered this in cargo's doc:
```
*-sys Packages

To alleviate linking to system libraries, Cargo has a convention of package naming and functionality. Any package named foo-sys will provide two major pieces of functionality:

The library crate will link to the native library libfoo. This will often probe the current system for libfoo before resorting to building from source.
The library crate will provide declarations for functions in libfoo, but it does not provide bindings or higher-level abstractions.
```

**So I think there is no way to make cargo to link to libpq.lib instead of pq.lib unless @sgrif rename pq-sys to libpq-sys???**

Versions:
cargo 0.16.0-nightly (d1bfc01 2016-11-18)
rustc 1.15.0-nightly (7b3eeea22 2016-11-21)
Windows 10 Version 1607 14393.447

Thanks!

**Comment by alexcrichton** [comment]
The name of a crate shouldn't affect the link flags at all, and in general this probably isn't a Cargo bug as Cargo's just ferrying information around. Are you sure that `pq.lib` shows up nowhere else in the codebase?

**Comment by fluxxu** [comment]
Sorry, my bad.
I missed a
```
#[link(name = "pq")]
```
😢
